runtime/compiler/env/j9method.cpp

```diff
@@ -5122,7 +5122,10 @@ TR_ResolvedJ9Method::methodModifiers()
 bool
 TR_ResolvedJ9Method::isConstructor()
    {
-   return nameLength()==6 && !strncmp(nameChars(), "<init>", 6);
+   if (!TR::Compiler->om.areValueTypesEnabled())
```

Comment:
do we need to case this? I would have thought these tests would always work for constructors? It might make sense from a compile-time perspective to do something like the strncmp last?

Reply:
>  It might make sense from a compile-time perspective to do something like the strncmp last?

strncmp should be moved to last.  

> do we need to case this? I would have thought these tests would always work for constructors?

It'd work. However when value type is disabled, if the method is a constructor, the check on returnType and isStatic will always return true and then strncmp is performed. With checking on `areValueTypesEnabled()`, the extra checks on returnType and isStatic could be avoided.